xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -1454,6 +1454,21 @@ time_t CPVRClients::GetPlayingTime() const
   return time;
 }
 
+bool CPVRClients::IsRecordable(const EPG::CEpgInfoTag *tag) const
+{
+  PVR_CLIENT client;
+  bool isRecordable = false;
+
+  if (GetClient(tag->ChannelTag()->ClientID(), client))
+  {
+     if(client->IsRecordable(tag, &isRecordable) != PVR_ERROR_NO_ERROR) {
+       isRecordable = tag->EndAsLocalTime() > CDateTime::GetCurrentDateTime();
```

Comment:
isRecordable is passed as "return"-parameter to IsRecordable and set there in case of PVR_ERROR_NO_ERROR (the PVR-addon does the decision). In all other cases, the previous behaviour of Kodi is used as this is probably what most PVR addons require.